app/classifier/tasks/text/index.cjsx

```diff
@@ -93,3 +79,50 @@ module.exports = React.createClass
       value = textBefore + startTag + textInBetween + endTag + textAfter
     newAnnotation = Object.assign @props.annotation, {value}
     @props.onChange newAnnotation
+
+  handleClear: (e) ->
+    # keyCode 8 is backspace, 46 is delete
+    if e.which is 8 or e.which is 46
+      @setState rows: 1
+
+  handleResize: ->
```

Comment:
handleResize could be simplified to:
```
    heightChange = @refs.textInput.scrollHeight - @refs.textInput.offsetHeight + 2
    return unless heightChange > 0
    rows = @state.rows
    switch heightChange
      when 44, 30
        rows = rows + 2
      when 66, 52
        rows = rows + 3
      when 88, 75
        rows = rows + 4
      else
        rows = rows + 1
    @setState rows: rows
```
but, then the (Chrome, Safari) edge case where there are 6+ rows of text and one character is deleted the textarea would be resized to 2 rows, instead of staying at 6+ rows (which is/will-be the case with Firefox). simply typing any additional character will properly resize the textarea back to 6+ rows, but immediately after deleting a single character, before typing another character, the textarea will be smaller sized than content.

i don't think the extra 3? lines of code are too bad, though it does make things less readable, but open to suggestions.